ui/app/pages/permissions-connect/permissions-connect.component.js

```diff
@@ -135,19 +135,22 @@ export default class PermissionConnect extends Component {
       return history.push(DEFAULT_ROUTE)
     }
 
-    if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN || getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
+    const environmentType = getEnvironmentType()
+    if (
+      environmentType === ENVIRONMENT_TYPE_FULLSCREEN ||
+      environmentType === ENVIRONMENT_TYPE_NOTIFICATION
+    ) {
       window.addEventListener('beforeunload', this.beforeUnload)
     }
   }
 
-  cancelPermissionsRequest = async (requestId) => {
+  cancelPermissionsRequest = (requestId) => {
     const { history, rejectPermissionsRequest } = this.props
     if (requestId) {
-      await rejectPermissionsRequest(requestId)
-
-      if (getEnvironmentType() === ENVIRONMENT_TYPE_FULLSCREEN || getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
+      rejectPermissionsRequest(requestId)
+      if (getEnvironmentType() === ENVIRONMENT_TYPE_NOTIFICATION) {
         window.close()
-      } else if (getEnvironmentType() === ENVIRONMENT_TYPE_POPUP) {
+      } else {
         history.push(DEFAULT_ROUTE)
```

Comment:
There's no reason to `await` any work in this function, nor for it to be `async`. It makes the fullscreen behavior more responsive, and it doesn't affect the popup.

Reply:
Yeah, that's a scary thought, and it didn't actually make much difference for the fullscreen behavior with the other changes I've made. Reverted.